### PR TITLE
Flesh out contribtuting instructions + Linting

### PR DESCRIPTION
**Pull request by OpenBagTwo**
This PR contains three commits which can be cherry-picked separately:

## Development Cruft

a9281631238c44a32ac8068dcc89e99efacac7cf brings over a conda environment YAML and a pre-commit config designed to make it easier for contributors to work on this project and keep their code well-formatted.

## Instructions for Obtaining Vanilla Assets

[The instructions right now](https://github.com/slicedlime/seasons/blob/1b57921a047c2ffb6a7ba474f3bc46dc34507ba5/CONTRIBUTING.md#L16-18) are rather minimal:

> It requires some vanilla files to derive the default values from. Create a 'vanilla' folder and
> copy the default 'foliage.png' and 'grass.png' files from the vanilla resource pack into it. Then
> copy the entire 'worldgen/biome' folder from the vanilla data pack into the folder as 'biome'.

I'm not sure if you're allowed to update the assets in your [`examples` repo](https://github.com/slicedlime/examples), but assuming you're not, I've done my best to give contributors explicit instructions to obtain them legally via 19c261f + fb2bea2. If you choose to leave off the environment YAML above, then the references to a conda environment should be purged or replace with a one-line environment creation command (_e.g._ `mamba create -n seasons python=3.11 imageio`)

## Linting (auto and otherwise)

Using the pre-commit configuration from the first commit, I auto-linted the two python modules and then resolved the remaining `mypy` errors, resulting in 45311d7. If you're disposed to accept these changes, **I strongly urge you to look over the diff** for `build.py` to make sure that the changes match with your intent. ~~as the datapack generated after this commit **does not match** the datapack generated prior to it. I suspect I know which line resulted in the change and will add a comment highlighting it.~~

By ways of validation, I deleted all the worldgen JSON files (`rm data/seasons/worldgen/biome/*/*.json`) theen rebuilt the datapack using `python build.py` on my HEAD, and the freshly-generated files were identical to what you currently have at a5b3663aa8e4c262ccf79f0d5f678c8dde678a06

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -244,25 +209,30 @@ def create_tags(id, biome, winter_biomes: list, bare_winter_biomes: list, snowy_
...
-    r = r * 255
-    g = g * 255
-    b = b * 255
-    return ((int(r) & 0xff) << 16) | ((int(g) & 0xff) << 8) | (int(b) & 0xff)
 
-def calculate_color(downfall: float, temperature: float, colormap: numpy.typing.NDArray) -> int:
+def rgb_to_int(r: float, g: float, b: float) -> int:
+    r = int(r) * 255
+    g = int(g) * 255
+    b = int(b) * 255
```
Comment (OpenBagTwo) [inline build.py:229]: So this is the line with the intent change, and it prompted me to have a look at the actual diffs:

```diff
diff -r post/seasons/worldgen/biome/winter_snowy/windswept_hills.json pre/seasons/worldgen/biome/winter_snowy/windswept_hills.json
18c18
<     "sky_color": 0,
---
>     "sky_color": 10925528,
21c21
<     "grass_color": 255,
---
>     "grass_color": 16056063,
```

And--yup, I can see what's going on and that the *type hint* was the part that needed to be fixed. Please hold.

File: data/seasons/worldgen/biome/spring_flowering/<built-in function id>.json [hunk 2]
```diff
@@ -0,0 +1,198 @@
+{
```
Comment (OpenBagTwo) [inline data/seasons/worldgen/biome/spring_flowering/<built-in function id>.json:1]: :thinking:

**Comment by slicedlime** [comment]
Well this seems to be doing 3 things:
- Completely re-writing the code style
- Adding a development environment of personal choice
- Adding some documentation on how to get the vanilla files

Of those, only the third part is of interest really, and even then I disagree that telling people that they can go grab the vanilla assets from some random resource pack is an ok way to build the pack. My suggestion would be to do one thing per pull request.

**Comment by OpenBagTwo** [comment]
I'll close this for now then in favor of a separating out those three workstreams.

Regarding this PR

> Completely re-writing the code style

I want to stress that the bulk of this was simply autoformatting applied by black and isort to make your code compliant with PEP8. It's, of course, your choice if you'd prefer not to adopt those standards, but I have found with my own work that they help a lot with legibility.

**Pull request closed without merging**